src/server/worker/pipeline/transform/registry.go

```diff
@@ -386,12 +390,12 @@ func (reg *registry) sendDatumTasks(ctx context.Context, pj *pendingJob, numDatu
 	maxBytesPerTask := int64(chunkSpec.SizeBytes)
 	driver := pj.driver.WithContext(ctx)
 	var numTasks int64
-	if numDatums < reg.concurrency {
+	if numDatums < reg.concurrency*taskGranularity {
 		numTasks = numDatums
 	} else if maxDatumsPerTask > 0 && numDatums/maxDatumsPerTask > reg.concurrency {
 		numTasks = numDatums / maxDatumsPerTask
 	} else {
-		numTasks = reg.concurrency
+		numTasks = reg.concurrency * taskGranularity
```

Comment:
I'm not sure this change is doing what you expect.  Imagine a pipeline with `concurrency=20` and `maxDatumsPerTask=3`.  If I've read this correctly, we have the following behavior before vs after:

| datums | tasks (before) | tasks (after) |
| -------- | --------------- | ------------- |
| 10 | 10 | 10 |
| 20 | 20 | 20 |
| 40 | 20 | ***80*** |
| 80 | 26 | 26 |
| 160 | 53 | 53 |

Note that with increasing datum numbers, the code before has a smooth transition in number of tasks with increasing number of datums - the new code has a discontinuity in the `else` case where the number of tasks spikes, then goes back down in the `else if` case.

Reply:
@Tryneus alright I thiiink I fixed this. It's a little hard to reason about but I think the issue was basically due to updating the default concurrency but not the `numDatums/maxDatumsPerTask > reg.concurrency` part of the conditional I was increasing the number of datums in some cases that I didn't mean to. I just change the condition to `numDatums/maxDatumsPerTask > reg.concurrency*taskGranularity` which is the equivalent of what it was before. Lmk if this pass the Marc test though.

Reply:
yes, this would do it